Assets/Scripts/Battlefield/AStar/AStar.cs

```diff
@@ -30,12 +30,12 @@ public Node(Node parent, int x, int y)
     public class AStar
     {
 
-        public List<Tile> FindPath(Tile goal, GridHolder grid, UniqueCreature creature)
+        public List<Tile> FindPath(Tile goal, GridHolder grid, MovementSystem ms)
```

Comment:
Closer, but what I meant was to have AStar and the line renderer inside the MovementSystem instead of a MovementSystem inside AStar. It would indicate a MovementSystem that uses and Astar Algorithm for movement and a LineRenderer to do movement.